src/kernel.c

```diff
@@ -1233,6 +1247,8 @@ mrb_init_kernel(mrb_state *mrb)
   mrb_define_method(mrb, krn, "to_s",                       mrb_any_to_s,                    MRB_ARGS_NONE());    /* 15.3.1.3.46 */
   mrb_define_method(mrb, krn, "__case_eqq",                 mrb_obj_ceqq,                    MRB_ARGS_REQ(1));    /* internal */
 
+  mrb_define_method(mrb, krn, "class_defined?",             mrb_krn_class_defined,           MRB_ARGS_REQ(1));
```

Comment:
It seems this method is unneeded because we can use `Object.const_defined?`.

Reply:
I thought it checks constant existence **and** if it's a class.
We have two options:
* use `const_defined?` instead of `class_defined?`
* make `class_defined?` checks if it's a class

Reply:
\# Sorry, I mistook place for comment.

I vote for using `const_defined?` for the following reasons:

- Ruby doesn't have `class_defined?`
- I think `class_defined?` doesn't need for core since it is used almost only for tests
- I think class or not isn't so important since program doesn't work fine anyway if constant isn't class

#### Off topic:

I think the following code isn't so good in the first place because skipping isn't shown in test results.

~~~ruby
assert('String#% with nan') do
  assert_equal "NaN", "%f" % Float::NAN
end if class_defined?("Float")
~~~

I believe `skip` should be used, for example:

~~~ruby
#
# equivalent to:
#
#   assert('String#% with nan') do
#     unless Object.const_defined?(:Float) && Object.const_get(:Float).kind_of?(Class)
#       skip "Float class isn't defined"
#     end
#     assert_equal "NaN", "%f" % Float::NAN
#   end
#
assert('String#% with nan', required_class: :Float) do
  assert_equal "NaN", "%f" % Float::NAN
end
~~~

Reply:
I removed `Kernel#class_defined?`.